### ISSUE TITLE
observables where no output is generated

### ISSUE THREAD

**Issue by wkotlarski** [FlexibleSUSY/FlexibleSUSY#173]
Is it correct that the calculation of observables in `run_MODEL.cpp` happens even if no spectrum is generated? It seems that there's no safeguard.

**Comment by Expander** [comment]
You are right, this is bad.

What about testing for `calculate_sm_masses` and `calculate_bsm_masses`?

**Comment by wkotlarski** [comment]
Your the expert but I think it's not a good idea. The problem I encountered was - imagine a point which fails, because a coupling gets non-perturbative or something. The spectrum is not calculated but the decays are and then the code complains because masses are 0 as they were never initialized. So even if  `calculate_sm_masses = true` this will still fail.

**Comment by Expander** [comment]
You are right! Then we should extend the logic and do something like the following (pseudo-code):
~~~
if (forceOutput || (no_problem && calculate_sm_masses && calculate_bsm_masses)) {
   calculate_observables();
}
~~~

**Comment by wkotlarski** [comment]
Is everyone ok with solution #175?

**Comment by Expander** [comment]
Please see my comment on PR #175.

**Comment by Expander** [comment]
Can this issue be closed, since PR #175 has been merged?
